src/utils/FileDownloader.ts

```diff
@@ -0,0 +1,94 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from "fs-extra";
+import * as path from "path";
+
+import axios, { AxiosRequestConfig, AxiosResponse } from "axios";
+import { Readable } from "stream";
+import { v4 as uuid } from "uuid";
+import { ExtensionContext, Uri, window } from "vscode";
+import { localize } from '../localize';
+
+export class FileDownloader {
+
+    public getDownloadStoragePath(context: ExtensionContext): string {
+        return path.join(context.globalStorageUri.fsPath, `file-downloader-downloads`);
+    }
+
+    public getDownloadedFile(fileName: string, context: ExtensionContext): Uri {
+        const downloadStoragePath = this.getDownloadStoragePath(context);
+        return Uri.file(path.join(downloadStoragePath, fileName));
+    }
+
+    public async downloadExternalBinary(url: string, fileName: string, context: ExtensionContext): Promise<Uri> {
+
+        const options: AxiosRequestConfig = {
+            timeout: 20000,
+            responseType: `stream`,
+            headers: {
+                Accept: "application/octet-stream",
+                "Content-Type": "application/octet-stream"
+            },
+            proxy: false // Disabling axios proxy support allows VS Code proxy settings to take effect.
+        };
+
+        const downloadStoragePath: string = this.getDownloadStoragePath(context);
+        await fse.ensureDir(downloadStoragePath);
+
+        // Generate a temporary filename for the download
+        const tempFileDownloadPath: string = path.join(downloadStoragePath, uuid());
+        const fileDownloadPath: string = path.join(downloadStoragePath, fileName);
+
+        const writeStream: fse.WriteStream = fse.createWriteStream(tempFileDownloadPath);
+
+        await axios.get(url, options)
```

Comment:
can we write `let response = await axios.get(url, options)` to avoid combining async/await and .then

Reply:
I tried to rewrite this without the await and it sometimes fails. This whole file will be replaced when my PR gets merged here: - https://github.com/microsoft/vscode-file-downloader/pull/42. I will then rewrite the code to use this other extension.

Reply:
This whole section has been replaced by the vscode-file-downloader extension